modules/core/include/opencv2/core/types.hpp

```diff
@@ -2017,6 +2017,15 @@ double jaccardDistance(const Rect_<_Tp>& a, const Rect_<_Tp>& b) {
     return 1.0 - Aab / (Aa + Ab - Aab);
 }
 
+/** @brief Finds out if there is any intersection between two rectangles
+ * 
+ * mainly useful for language bindings
+ * @param rect1 First rectangle
+ * @param rect2 Second rectangle
+ * @return the area of the intersection
+ */
+CV_EXPORTS_W inline double rectangleIntersection(const Rect2d& a, const Rect2d& b) { return (a & b).area(); }
```

Comment:
Shouldn't we also add this function for `cv::Rect`? Will there be a conflict if the have the same name?

Reply:
> rectangleIntersection

Name is not accurate. It returns "area".

---

There is no `cv::Rect` automatic wrapper in bindings.
We could add helper Python class for Rect support here: https://github.com/opencv/opencv/tree/4.x/modules/core/misc/python/package

Reply:
There is a wrapper for `cv::Rect` in common code: https://github.com/opencv/opencv/blob/0ac67ee8d92e3fef8bb78e774cc004b8cdba180e/modules/python/src2/cv2_convert.cpp#L674-L687

Reply:
> Shouldn't we also add this function for cv::Rect? Will there be a conflict if the have the same name?

currently int rects will be converted to double by parseSequence, so they do work. An int overload would be handled by the wrapper just fine, but I doubt there would be any performance benefits (using python)

> Name is not accurate. It returns "area".

tried to follow rotatedRectangleIntersection here, which also returns the intersection area. Then there is intersectConvexConvex. What name would you suggest? rectangleIntersectionArea sounds rather verbose..

Reply:
@alalek the naming is the only outstanding issue here, right? Any suggestions?